### DISCUSSION TITLE
Is this setup possible with coolify?

### DISCUSSION THREAD

**Discussion by isaacfink** [coollabsio/coolify#2902] [Q&A]
I played around with coolify a bit over the weekend and it looks amazing so far, huge thanks to the creator for an awesome project, however one complaint I do have is for someone like me with minimal devops experience I could benefit from more templates and project setups, I have a specific setup in mind and I couldn't find any information on how to achieve some of the functionality

I want a setup hat could easily create staging and dev environments, each environment should automatically set the environment variables to the relative containers, for example if I add a postrges database for each environment I want my staging application to point to this environment by default, I also want the database data to persist between deployments, same for minio

There are more variables that need to work this way, I also want the deployment urls to all be the same but under a different url, for example staging should be staging.example.cloud and it should have a variable for the local url for the database that belongs to this instant, the admin site should be admin.staging.example.com but without me having to specify it for each deployment, ideally this would come from the branch name on pull requests but it could be anything as long as I only have to specify it once (like using the env name), I saw coolify adds environment variables with the info I need but how can I use that in the url?

It would also be great if I can add a coolify config to my repo this way I could do all that automatically and perform any changes in code and not in the UI

**Comment by mirsella** [comment]
update it was not working because of the stripprefix option
